### DISCUSSION TITLE
Vendure Storefront

### DISCUSSION THREAD

**Discussion by zohra-alt** [vendurehq/vendure#805] [Q&A]
I want to ask If it is possible, how I can use Next.js Commerce as a storefront with vendure. I didn't find any documentation, explanation, or tutorial about that?
Next.js: https://nextjs.org/commerce
Vendure: https://www.vendure.io/

**Comment by Izayda** [comment]
Hi! We use Next.js as storefront (dev stand https://devroooms.ru/). What questions do you have?

**Reply by zohra-alt** [reply]
I am trying to integrate Vendure with Next.js Commerce, but I find that Next.js Commerce works with Shopify and Bigcommerce. I just want the frontend store template without any of these two providers, integrated with Vendure. And there is no clear documentation about that, neither in the Vendure documentation nor in the Next.js Commerce documentation. 

Is that possible?
If yes, how can I do that? Where can I find some tutorial or documentation about that?

**Reply by Izayda** [reply]
Sorry, i read your question inattentively first time. We use pure Next.js, we don't use Next.js Commerce

**Reply by zohra-alt** [reply]
Okay, thank you anyway.

**Reply by zecsn** [reply]
> Hi! We use Next.js as storefront (dev stand https://devroooms.ru/). What questions do you have?

Hi, I also want to use Next.js for front store. But I have some questions

Are you using a complete separate project or is inside plugins/ui-extention directory?

What are the settings for Next.js (build, port, etc)?

**Reply by Izayda** [reply]
Hi! It's complete separate project. Backend and Frontend now are hosted on the same machine behind nginx. Frontend communicate with backend by devroooms.ru/shop-api
Nginx is also configured to redirect to domain, if is asked by ip. 
Frontend is packaged to docker image, backend is deployed directly. 

It'a dev env, for prod env the both will be packaged to docker image, probably on separate machines.

**Comment by michaelbromley** [comment]
Hi! 

We have an open PR to add support for Vendure to Next.js Commerce: https://github.com/vercel/commerce/pull/223
Feel free to comment/like the related issue: https://github.com/vercel/commerce/issues/220, which might help get more attention on getting the PR looked at.

**Reply by zohra-alt** [reply]
Thank  You. That was really helpful.
